extra_foam/gui/image_tool/tests/test_image_tool.py

```diff
@@ -382,6 +382,23 @@ def testReferenceCtrlWidget(self):
         self.assertTrue(updated)
         self.assertIsNone(ref)
 
+        # test recording reference
+        # setup initial corrected image
+        image = np.ones((10, 10)).astype(np.float32)
+        # start recording
+        QTest.mouseClick(widget.record_btn, Qt.LeftButton)
+        # receive 5 trains
+        for received_train in range(1, 6):
+            corrected._image = image * received_train
+            # mock view update as it contains the averaging logic
+            view.updateF(ProcessedData(1), False)
+        # stop recording
+        QTest.mouseClick(widget.record_btn, Qt.LeftButton)
+        # retrieve reference value and verify it is the average of the trains
+        updated, ref = proc._ref_sub.update()
+        self.assertTrue(updated)
+        np.testing.assert_array_equal(image * 3, ref)
```

Comment:
Excuse the naive question, but why do you assert for ``image * 3``, after receiving 5 trains?

Reply:
```python
ref = (image + 2*image + 3*image + 4*image + 5*image ) /  5

```

Reply:
Thanks Ebad, I see, like 3 being the mean (expectation value) in the series 1-5, sure thing 😄